### PR TITLE
Add upper lower bounds

### PR DESCRIPTION
**Pull request by xiaohui-sun**
Add upper/lower bounds to TimeSeries.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java [hunk 1]
```diff
@@ -53,17 +53,47 @@ public void addPredictedBaseline(DoubleSeries baselineValues) {
     this.df.addSeries(DataFrameUtils.COL_VALUE, baselineValues);
   }
 
+  /**
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java]: Shouldn't the result include current time series too?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java [hunk 2]
```diff
@@ -53,17 +53,47 @@ public void addPredictedBaseline(DoubleSeries baselineValues) {
...
+   * Add the predicted upper bound into the timeseries
+   * @param upperBoundValues predicted upper bound values
+   */
+  public void addPredictedUpperBound(DoubleSeries upperBoundValues) {
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java]: I'd prefer we have a time series builder class to construct the TimeSeries class. Something like:

` Public static class Builder() { `
        `  public Builder(LongSeries timeStamp, DoubleSeries currentTimeSeries, DoubleSeries predictedBaseline); `
       `   public Builder upperBound(DoubleSeries predictedUpperBound);`
       `   public Builder lowerBound(DoubleSeries predictedLowerBound);`
       ` .  public TimeSeries Build();`
`   }
`
Then the DataFrame class won't be exposed to client and we have a explicit contract to build TimeSeries instance.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java (hunk 1 above)
Comment (xiaohui-sun) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java]: Good point! We should include current as well.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java (hunk 2 above)
Comment (xiaohui-sun) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java]: As discussed. Changed the constructors to handle it more efficiently.

**Pull request merged**